src/native/fs/fs_napi.cpp

```diff
@@ -856,19 +898,33 @@ struct Rename : public FSWorker
 struct Writefile : public FSWorker
 {
     std::string _path;
+    XattrMap _xattr;
+    bool _set_xattr;
+    bool _xattr_need_fsync;
+    std::string _xattr_clear_prefix;
     const uint8_t* _data;
     size_t _len;
     mode_t _mode;
     Writefile(const Napi::CallbackInfo& info)
         : FSWorker(info)
+        , _set_xattr(false)
         , _mode(0666)
     {
         _path = info[1].As<Napi::String>();
         auto buf = info[2].As<Napi::Buffer<uint8_t>>();
         _data = buf.Data();
         _len = buf.Length();
         if (info.Length() > 3 && !info[3].IsUndefined()) {
-            _mode = info[3].As<Napi::Number>().Uint32Value();
+        }
+        if (info.Length() > 3 && !info[3].IsUndefined()) {
```

Comment:
is this cond `if (info.Length() > 3 && !info[3].IsUndefined()) { } ` appears twice?

Reply:
fixed, thanks